### ISSUE TITLE
Create pagination in the BE

### ISSUE THREAD

**Issue by B-Naoufal** [massalabs/station#1004]
**Context**
Our code currently experiences performance issues when retrieving and displaying a large number of DNS entries. One particular bottleneck is the DNSRecordFavicon function, which takes at least 20 ms to fetch the favicon for each website. With over 800 DNS entries, this results in significant delays.
**Solution**
To improve performance, we propose implementing pagination. By fetching and displaying only a limited number of entries at a time, such as 10 per page, we can reduce the latency of search pages to around 200 ms. When users navigate to the next page, we fetch the corresponding data for the next set of 10 websites.   
Additionally, another solution is to selectively display favicons for some websites. Favicon fetching can be a time-consuming process, contributing to the overall latency.
  
For more details check the implementation of [Registry()](https://github.com/massalabs/station/blob/c87375cc8d5cfd94be3ab5466c56edc53d1881ce/int/api/websites/all.go#L56C45-L56C51)  and the part of extracting the [favicon](https://github.com/massalabs/station/blob/c87375cc8d5cfd94be3ab5466c56edc53d1881ce/int/api/websites/all.go#L95C5-L95C5) for each dns.*

**User flow**
*Search Page should display in  acceptable time with relevant data -PM-?*

**How to**
*Change the implementation of Registry Function to render only limited number of websites at the first page + available pages  numbering*

**Technical details**

1. *Update the endpoint  response [/all/domains](https://github.com/massalabs/station/blob/c87375cc8d5cfd94be3ab5466c56edc53d1881ce/api/swagger/server/restapi/resource/swagger.yml#L14C12-L14C12)*.   
2. Modify the handler function signature: Update the handler function signature to accept parameters for pagination.  
3. Modify the Registry function to accept the page number and entries per page as parameters.  
4. Implement pagination logic.  
5. Adapt the front end to display the pagination shortcuts.

**QA testing**
*Does this task require some QA tests ?*
*If yes, explain how to validate it*

**Comment by peterjah** [comment]
Pagination is nice to have, but maybe the most efficient way to reduce the loading time would be to fetch all favicons in parallels. Is it possible ?

I don't undersant the link with configure_acrylic.bat

**Comment by gregLibert** [comment]
One temporary and immediate solution could be to also clean the DNS from all the unused website. How is it possible that we are having 800 Entries ?

**Comment by B-Naoufal** [comment]
> One temporary and immediate solution could be to also clean the DNS from all the unused website. How is it possible that we are having 800 Entries ?

At Each ci we run robot tests which test all endpoints including the one to upload a website ( add new dns entry ) so each push -> new entry added to the dns

**Comment by mazmassa** [comment]
> Pagination is nice to have, but maybe the most efficient way to reduce the loading time would be to fetch all favicons in parallels. Is it possible ?
> 
I guess pagination will give us a much better experience. Indeed. Once done if we still suffer with that we can think in a lazy load for that indeed. I would suggest one piece flow. Pagination, if is still an issue we can jump to lazy load for that. WDYT!?

**Comment by SlnPons** [comment]
@B-Naoufal what about this point raised by Greg please? Haven check during your analysis? https://github.com/massalabs/station/issues/1004#issuecomment-1633853896

@mazmassa I'm not sure to fully understand your comment, sorry :/. The point raised by Pierre, and with which I agree, is that creating pagination is expensive so it's a nice to have. And so fetching the favicons in parallel or in an another way could improve this loading time without being as expensive to build as the pagination was the question.
And you're saying: "I suggest a one flow", ie: fetching favicons in parallel isn't a good solution. Is that correct?

And finally, @gregLibert should we include an improvement regarding the tests to avoid spamming the buildnet and DNS SC at each push?

**Comment by gregLibert** [comment]
> And finally, @gregLibert should we include an improvement regarding the tests to avoid spamming the buildnet and DNS SC at each push?

Yes
